lib/ruby_lsp/document.rb

```diff
@@ -32,6 +32,13 @@ def ==(other)
       @source == other.source
```

Comment:
Why don't we call `parse` here instead? I think we'd always want to parse it?

Reply:
Sorry I meant parse in `Document#initialize` like we currently do. This PR does 2 things:

1. Extract and postpone document parsing
2. Rescue the parsing error

And I'm not sure why we need to do 1 because it adds another operation we need to keep in mind when dealing with documents.

Reply:
Oh, I see what you mean. Indeed, we don't need the `parse` method. I've moved rescuing inside `initialize` and simplified these changes.